test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -517,10 +536,28 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"run-test-unit\n", returnData.stdOut);
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"-test-\n", returnData.stdErr);
     }
+    
+    void TestDoInvokeMethodScriptTmpDir()
+        {
+            std::wstring errMsg;
+            mi::SCX_OperatingSystem_ExecuteScript_Class param;
+            param.Script_value(
+            		            "#!/bin/sh\n"
+            		            "# Get location pointed by tmpdir\n"
+            					"DIR=`dirname $0`\n"
+            		            "echo $DIR\n"
+            		            "exit 0\n");
+            param.Arguments_value("unit test run");
+            param.timeout_value(0);
+            InvokeReturnData returnData;
+            ExecuteScript(param, MI_RESULT_OK, returnData, CALL_LOCATION(errMsg));
+            CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, 0, returnData.returnCode);
+            CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"./tmpdir\n", returnData.stdOut);
+        }
```

Comment:
Sure would like to see an additional test to test in a non-standard location. You're not really testing the entire point of this change; allowing tmpdir to be relocated.

You should probably have three tests:

1) One if tmpdir doesn't exist at all (install issue or something); better to work with /tmp than to fail completely, although it's reasonable to log a one-time warning in this case,

2) One with tmpdir set to /tmp/, and

3) One with tmpdir set to some special location that you create.

By doing this, you actually test the feature you want. But if you do that, then it likely makes sense that tearDown() shouldn't do the cleanup, since the names can vary, and the cleanup might vary (i.e. delete tmpdir and delete the directory that it points to).